source/tutorials/installing-nixos-on-a-raspberry-pi.rst

```diff
@@ -0,0 +1,184 @@
+
+You should be greeted with a fresh shell!
+
+
+Getting internet connection
+---------------------------
+
+Run ``sudo -i`` to get a root shell for the rest of the tutorial.
+
+At this point we'll need internet connection. If you can use an ethernet cable, plug it in.
+
+In case you're connecting to a wifi run ``iwconfig`` to see what is the name of your wireless
+network interface. In case it's ``wlan0`` replace ``SSID`` and ``passphrase`` with your data and run:
+
+.. code:: shell-session 
+
+  # wpa_supplicant -B -i wlan0 -c <(wpa_passphrase 'SSID' 'passphrase') &
+
+
+Once you see in your terminal that connection is established, run ``host google.com`` to 
+check that DNS resolves correctly.
+
+In case you've made a typo, run ``pkill wpa_supplicant`` and start over.
+
+
+Updating firmware
+-----------------
+
+To benefit from updates and bug fixes from the vendor, we'll start by updating Raspberry Pi firmware:
+
+.. code:: shell-session
+
+  # nix-shell -p raspberrypi-eeprom
+  # FIRMWARE_RELEASE_STATUS=stable rpi-eeprom-update -d -a
+
+  
+Installing NixOS 
+----------------
+
+For initial installation we'll install `XFCE <https://www.xfce.org/>`_ desktop environment
+with user ``guest`` and SSH daemon.
+
+.. code:: nix 
+
+  { config, pkgs, lib, ... }:
+
+  let
+    user = "guest";
+    password = "guest";
+    SSID = "mywifi";
+    SSIDpassword = "mypassword";
+    interface = "wlan0";
+    hostname = "myhostname";
+  in {
+    imports = ["${fetchTarball "https://github.com/domenkozar/nixos-hardware/archive/rpi4.tar.gz" }/raspberry-pi/4"];
+
+    fileSystems = {
+      "/" = {
+        device = "/dev/disk/by-label/NIXOS_SD";
+        fsType = "ext4";
+        options = [ "noatime" ];
+      };
+    };
+
+    networking = {
+      hostName = hostname;
+      wireless = {
+        enable = true;
+        networks."${SSID}".psk = SSIDpassword;
+        interfaces = [ interface ];
+      };
+    };
+
+    environment.systemPackages = with pkgs; [ vim ];
+
+    services.openssh.enable = true;
+
+    users = {
+      mutableUsers = false;
+      users.guest = {
+        isNormalUser = true;
+        password = password;
+        extraGroups = [ "wheel" ];
+      };
+    };
+
+    # video
+    services.xserver = {
+      enable = true;
+      displayManager.lightdm.enable = true;
+      desktopManager.xfce.enable = true;
+      videoDrivers = [ "fbdev" ];
+    };
+
+    # audio
+    hardware.pulseaudio.enable = true;
+  }
+
+To save time on typing the whole configuration, download it:
+
+.. code:: shell-session
+
+  # curl -L https://tinyurl.com/nixos-rpi-tutorial-preview > /etc/nixos/configuration.nix 
+
+At the top of `/etc/nixos/configuration.nix` there are a few variables that you want to configure,
+most important being your wifi connection details, this time specified in declarative way.
+
+Once you're ready to install NixOS:
+
+.. code:: shell-session
+
+  # nixos-install --root /
```

Comment:
Just a note, nothing to change. A `nixos-rebuild` would work just as well... Except that `nixos-install` prompts you to set a root password, and could be doing other "housekeeping" stuff.

Reply:
Let's keep it as-is for now, although I'm tempted by the simplification. Maybe it's still better to teach how NixOS is installed uniformly.